### ISSUE TITLE
Incorrect onTapHint being applied 

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#62073]
b/161765861

This seems like a regression since onTapHint seemed to work correctly before. Both cases produces different results, but the issue should be related.

## Case 1

```dart
Semantics(
  button: true,
  onTap: () {},
  onTapHint: 'onTapHint',
  child: Container(color: Colors.blue, width: 100, height: 100, child: Text('text')),
)
```

Expected: TalkBack should read "text, Button" / "Double-tap to onTapHint."
Observed: TalkBack reads "text, Button" / "Double-tap to text."

## Case 2

```dart
Semantics(
  button: true,
  onTap: () {},
  onTapHint: 'onTapHint',
  child: Container(color: Colors.blue, width: 100, height: 100),
)
```

Expected:

TalkBack should read "Button" / "Double-tap to onTapHint."
Observed: App crashes with the following stack trace:

```
[ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)]
java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
[   +7 ms] E/flutter (17057): 	at io.flutter.view.AccessibilityBridge.updateCustomAccessibilityActions(AccessibilityBridge.java:1264)
[   +4 ms] E/flutter (17057): 	at io.flutter.view.AccessibilityBridge$1.updateCustomAccessibilityActions(AccessibilityBridge.java:260)
[   +4 ms] E/flutter (17057): 	at io.flutter.embedding.engine.FlutterJNI.updateCustomAccessibilityActions(FlutterJNI.java:500)
[   +4 ms] E/flutter (17057): 	at android.os.MessageQueue.nativePollOnce(NativeMethod)
[   +3 ms] E/flutter (17057): 	at android.os.MessageQueue.next(MessageQueue.java:336)
[   +4 ms] E/flutter (17057): 	at android.os.Looper.loop(Looper.java:174)
[   +4 ms] E/flutter (17057): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
[   +4 ms] E/flutter (17057): 	at java.lang.reflect.Method.invoke(Native Method)
[   +3 ms] E/flutter (17057): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
[   +4 ms] E/flutter (17057): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
[   +4 ms] E/flutter (17057): 
[   +4 ms] F/flutter (17057):
[FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(1019)]
Check failed: CheckException(env). 
```

**Comment by mehmetf** [comment]
Unfortunately, this was a result of a g3 roll that contained 4 massive engine rolls.  FYI @goderbauer @justinmc

**Comment by goderbauer** [comment]
Looks related to https://github.com/flutter/flutter/issues/60926, the crash is the same.

**Comment by goderbauer** [comment]
@chunhtai is working on a fix for the crash: https://github.com/flutter/engine/pull/19623

**Comment by TahaTesser** [comment]
**Case 2**

<details> 
<summary>logs</summary> 

``` 
Launching lib/main.dart on SM M305F in debug mode...
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:50358/B_YM0aSMW0U=/ws
E/flutter ( 3862): [ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)] java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
E/flutter ( 3862): 	at io.flutter.view.AccessibilityBridge.updateCustomAccessibilityActions(AccessibilityBridge.java:1264)
E/flutter ( 3862): 	at io.flutter.view.AccessibilityBridge$1.updateCustomAccessibilityActions(AccessibilityBridge.java:260)
E/flutter ( 3862): 	at io.flutter.embedding.engine.FlutterJNI.updateCustomAccessibilityActions(FlutterJNI.java:500)
E/flutter ( 3862): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/flutter ( 3862): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/flutter ( 3862): 	at android.os.Looper.loop(Looper.java:197)
E/flutter ( 3862): 	at android.app.ActivityThread.main(ActivityThread.java:7807)
E/flutter ( 3862): 	at java.lang.reflect.Method.invoke(Native Method)
E/flutter ( 3862): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/flutter ( 3862): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1047)
E/flutter ( 3862):
F/flutter ( 3862): [FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(1019)] Check failed: CheckException(env).
F/libc    ( 3862): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 3862 (.master_flutter), pid 3862 (.master_flutter)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'samsung/m30ltedd/m30lte:10/QP1A.190711.020/M305FDDU5CTF2:user/release-keys'
Revision: '4'
ABI: 'arm64'
Timestamp: 2020-07-23 18:59:57+0530
pid: 3862, tid: 3862, name: .master_flutter  >>> com.example.master_flutter <<<
uid: 10335
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: '[FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(1019)] Check failed: CheckException(env).
'
    x0  0000000000000000  x1  0000000000000f16  x2  0000000000000006  x3  0000007fdc622eb0
    x4  0000000000000080  x5  0000000000000080  x6  0000000000000080  x7  8000000000000000
    x8  00000000000000f0  x9  b4c13c24148709a9  x10 0000000000000001  x11 0000000000000000
    x12 fffffff0fffffbdf  x13 0000000005a1baf8  x14 0000000000000000  x15 00005cc70d39f6ae
    x16 00000073230d78c0  x17 00000073230b52d0  x18 00000073263c4000  x19 0000000000000f16
    x20 0000000000000f16  x21 00000000ffffffff  x22 00000072a07755c8  x23 00000000ffffffff
    x24 0000000000000018  x25 0000000000000018  x26 00000000000000d6  x27 00000000000000bc
    x28 0000000000000031  x29 0000007fdc622f50
    sp  0000007fdc622e90  lr  000000732306a038  pc  000000732306a064
backtrace:
      #00 pc 0000000000083064  /apex/com.android.runtime/lib64/bionic/libc.so (abort+160) (BuildId: f0e764f03a5004a006cd457ee8a428dc)
      #01 pc 000000000125bac0  /data/app/com.example.master_flutter-agc9s7uhZZcwSmqLLX3UCg==/lib/arm64/libflutter.so (BuildId: 4176bb07918ad946d275e1431bc426eaffd6ad96)
      #02 pc 00000000000018cc  [anon:.bss]
Lost connection to device.
Exited (sigterm)

``` 

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.21.0-2.0.pre.95, on Mac OS X 10.15.6 19G73, locale en-GB)
    • Flutter version 1.21.0-2.0.pre.95 at /Users/tahatesser/Code/flutter_master
    • Framework revision 50b87dc6c3 (19 hours ago), 2020-07-22 11:09:59 -0700
    • Engine revision dcc9a4048d
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 9dca49e71e)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (5 available)
    • SM M305F (mobile)      • 32003c30dc19668f          • android-arm64  • Android 10 (API 29)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 13.6
    • macOS (desktop)        • macos                     • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)       • web-server                • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 84.0.4147.89

• No issues found!
```

</details>

**Comment by chunhtai** [comment]
I confirm this is the same issue as https://github.com/flutter/flutter/issues/60926
applying the fix solved both crash and on tap hint announcement

**Comment by chunhtai** [comment]
close as dup
